PackageBase._has_make_target fails if not English locale

For a package based on Makefiles or autotools, "spack install --test ..." searches for targets "test", then "check", respectively, after the build. This uses PackageBase._has_make_target which checks for error messages from "make -n <target>" on stderr. However, if LANG is e.g. set to "de_DE.UTF-8", the error is "make: *** No rule to make target 'test'.  Schluss.". This results in _has_make_target wrongly returning True.

With "unset LANG", it works as expected. But I think spack should unset LANG by itself as it does not check for the language specific error message of make.

### Steps to reproduce the issue

With German translations for make installed:

```console
$ export LANG=de_DE
$ spack install --test root valgrind
...
```

### Error Message

==> RUN-TESTS: build-time tests [check]
==> 'make' '-j24' '-n' 'test'
==> 'make' '-j24' 'test'
make: *** No rule to make target 'test'.  Schluss.
==> Error: ProcessError: Command exited with status 2:
    'make' '-j24' 'test'

Stack runs "make -j24 test" because it assumes that this target exists.
Due to that, it does not go on to test for target "check".

BTW: the correct target for running the test suite in Valgrind is "make regtests". I will to a PR to fix this in the upstream Valgrind spec file.

### Information on your system

SLES12SP3

--- a/lib/spack/spack/package.py
+++ b/lib/spack/spack/package.py
@@ -1139,7 +1139,10 @@ class PackageBase(with_metaclass(PackageMeta, PackageViewMixin, object)):
         Returns:
             bool: True if 'target' is found, else False
         """
-        make = inspect.getmodule(self).make
+        make = copy.deepcopy(inspect.getmodule(self).make)
+
+        # Use English locale for missing target message comparison
+        make.add_default_env('LC_ALL', 'C')
 
         # Check if we have a Makefile
         for makefile in ['GNUmakefile', 'Makefile', 'makefile']:

--- a/lib/spack/spack/package.py
+++ b/lib/spack/spack/package.py
@@ -1139,6 +1139,7 @@ class PackageBase(with_metaclass(PackageMeta, PackageViewMixin, object)):
         Returns:
             bool: True if 'target' is found, else False
         """
+        # Prevent altering LC_ALL for 'make' outside this function
         make = copy.deepcopy(inspect.getmodule(self).make)
 
         # Use English locale for missing target message comparison

